src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorParser.cs

```diff
@@ -50,7 +56,28 @@ public class MvcRazorParser : RazorParser
                 TagHelperDescriptorResolver,
                 _globalImportDirectiveDescriptors,
                 errorSink);
-            return visitor.GetDescriptors(documentRoot);
+
+            var descriptors = visitor.GetDescriptors(documentRoot);
+            foreach (var descriptor in descriptors)
+            {
+                foreach (var attributeDescriptor in descriptor.Attributes)
+                {
+                    if (attributeDescriptor.IsIndexer &&
+                        string.Equals(
+                            attributeDescriptor.TypeName,
+                            _modelExpressionTypeName,
+                            StringComparison.Ordinal))
+                    {
+                        errorSink.OnError(SourceLocation.Undefined, Resources.FormatMvcRazorParser_InvalidPropertyType(
+                            descriptor.TypeName,
+                            attributeDescriptor.Name,
+                            attributeDescriptor.TypeName,
```

Comment:
Will remove this parameter from the resource. This value passed here is no longer the property type and, though it would be easy to find in the `Attributes` collection today, the corresponding dictionary descriptor won't always be available in the future i.e. once we support get-only dictionary properties. Rather than guess `IDictionary<string, ModelExpression>`, I'll reword things now.